TFT/src/User/API/parseACK.c

```diff
@@ -908,19 +908,19 @@ void parseACK(void)
       {
         if (ack_seen("L:"))
         {
-          while ((dmaL2Cache[ack_index]<'0') || (dmaL2Cache[ack_index]>'9') || dmaL2Cache[ack_index] == "\n")
+          while ((dmaL2Cache[ack_index]<'0') || (dmaL2Cache[ack_index]>'9'))
```

Comment:
What if there's no numeric value after "L:"?

Reply:
In fact, it should be `'\n'`  not `"\n"`.  and  '\n' is 0x0A, It is smaller than ‘0’(0x30), so it directly meets the condition of `(dmaL2Cache[ack_index]<'0')`.  and  `dmaL2Cache[ack_index] == '\n'` will never be executed

Reply:
I am glad I can learn every day from here.
So, in the light of what you've told it should be:
`while (((dmaL2Cache[ack_index]<'0') || (dmaL2Cache[ack_index]>'9')) && dmaL2Cache[ack_index] != '\n')`
to avoid "ack_index" going over the legth of "dmaL2Cache".

Reply:
It's done now. please check again.
Very very thank you for your nice work!

Reply:
哪里哪里, 不用謝. 你太 客气了!